### ISSUE TITLE
No further splits with positive gain, best gain: -inf

### ISSUE THREAD

**Issue by iFe1er** [Microsoft/LightGBM#1026]
i met with 2 problem in training LightGBM on python.

Firstly:  
i was doing a regression with loss:fair, and metrics:mae.  
While i was run lgb.cv() , the program pop out the warning message once for a while:  such as

[1620]	cv_agg's l1: 0.598669
[1640]	cv_agg's l1: 0.598657
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[1660]	cv_agg's l1: 0.598639
[1680]	cv_agg's l1: 0.598635

however, it dont stop training, it just keep training. What happen here? am i giving too many num_leaves? my dataset is 120000row x 50col , and num_leaves is only 10 with max_depth 7.

Secondly:  
Since i want to do a regression on a specific regression problem, with its label integer vary from 0 to 100 or even larger, and the evaluation metrics is MAE. So i tried customize my metrics to a "rounded MAE"(as follows):

```
  from sklearn.metrics import mean_absolute_error as mae
def round_mae(preds, train_data):  
    labels = train_data.get_label()  
    return 'error', mae(labels,[round(i) for i in preds]), False  
```

and run the lgb.cv with feval=round_mae

however, this time, it immediately stop when the the cv output run into the warning message:

[LightGBM] [Warning] No further splits with positive gain, best gain: -inf  

and the training stop, far from the normal early stop.

What happen here? Why is metrics  "MAE" can continue but a "custom MAE" stop when no split can be made? is that possible i can fix it?

**Comment by chivee** [comment]
it should stopped with MAE, could you please paste all the parameters that you are using.

in addition, try to set `min_data_in_leaf=1` and rerun the experiments.
BTW, do you have statics of your label data, such as the mean and the variances, it's better to smooth the regression value first.

**Comment by guolinke** [comment]
@iFe1er 
That is just a warning. You can ignore it.
